subprojects/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/CrossCompilationIntegrationTest.groovy

```diff
@@ -46,18 +46,21 @@ class CrossCompilationIntegrationTest extends AbstractIntegrationSpec {
                 sourceCompatibility = JavaVersion.${version.name()}
             }
 
-            def javaInstallation = project.javaInstalls.installationForDirectory(project.layout.projectDirectory.dir("${jvm.javaHome.toURI()}"))
+            def launcher = javaToolchains.launcherFor {
+                languageVersion = JavaLanguageVersion.of(${version.majorVersion})
+            }.get()
+            def installationDirectory = launcher.metadata.installationPath
 
             tasks.named("compileJava") {
                 options.fork = true
-                options.forkOptions.javaHome = javaInstallation.get().installationDirectory.asFile
+                options.forkOptions.javaHome = installationDirectory.asFile
```

Comment:
On first sight, was going to ask about why not use toolchains there.
But when looking at the test overall, looks like we could simply get rid of it.

Reply:
Really? I think this is a fair integration test to test cross-compilation using fork options. This is still a supported behavior that we didn't deprecate.